src/PaymentBundle/Controller/PaymentController.php

```diff
@@ -26,6 +26,16 @@
 
 class PaymentController extends Controller
 {
+    /**
+     * @var BasketFactoryInterface
+     */
+    private $basketFactory;
+
+    public function __construct(BasketFactoryInterface $basketFactory = null)
+    {
+        $this->basketFactory = $basketFactory;
```

Comment:
I think we should also trigger a deprecation error here if `$basketFactory` is `null`

Reply:
you mean use something @trigger_error what kind of message should i put?

Reply:
Yes, there are tons of examples you can find in Sonata if you grep for `E_USER_DEPRECATED`. In your case,
```
@trigger_error(sprintf(
    'Not providing a %s instance to %s is deprecated since sonata-project/ecommerce 3.x. Providing it will be mandatory in 4.0',
    BasketFactoryInterface::class,
    E_USER_DEPRECATED
));
```